packages/runtime/container-runtime/src/summary/runningSummarizer.ts

```diff
@@ -556,6 +567,88 @@ export class RunningSummarizer implements IDisposable {
 		return resultsBuilder.build();
 	}
 
+	/**
+	 * Tries to summarize with retries where retry is based on the failure params.
+	 * For example, summarization may be retried for failures with "retryAfterSeconds" param.
+	 */
+	private async trySummarizeWithRetries(
+		reason: SummarizeReason,
+		cancellationToken: ISummaryCancellationToken,
+	) {
+		let maxAttempts = defaultMaxAttempts;
```

Comment:
In practice, the semantics of this are kind of weird, since it updates each time through the loop. So like you could fail to submit 3 times then fail after once and be done since max is suddenly 2.

I actually think it's fine, but warrants a brief comment explaining that it's hard to predict how many attempts will actually happen, depending on how far each attempt makes it through the process.